### ISSUE TITLE
Minio client: Possible bug in the manipulation of BPMN models

### ISSUE THREAD

**Issue by igormoreno** [benchflow/benchflow#259]
I think I may have found a bug.

In `BenchFlowTestMinioClient` there are CRUD-like operations on `TestBPMNModel` and there is also a function to `copyBPMNModelForExperiment(String testID, String experimentID, String modelName)` which essentially does `copyObject(testObjectID, experimentObjectID)`. So far so good. But…

In `BenchFlowExperimentMinioClient` the only operation on BPMN models is `getExperimentBPMNModel(String experimentID, String modelName)` which 1st generates a `testID` from the `experimentID` (with `String testID = experimentID.substring(0, experimentID.lastIndexOf("."));`) and then gets the object using the `testID`. That's weird right?

you can create a BPMN model for an experiment using the test BPMN model but you never retrieve that experiment BPMN model because when you ask for it you actually get the test BPMN model retrieved using a testID generated from the experimentID.

Besides that, if there is a relationship between a testID and an experimentID (given by `String testID = experimentID.substring(0, experimentID.lastIndexOf("."));`) I don't see that relationship being enforced anywhere. And there seems to be other places where that should be checked. I mean other places that manipulate both testID and experimentID.

**Comment by jeppe-style** [comment]
@igormoreno can you send a link to the code where you found this?

Regarding the relationship between testID and experimentID:
the general structure is: `<username>.<test name>.<test number>.<experiment number>`
so the difference between a testID and experimentID is basically the number of dots (2 or 3). I was thinking also to make this a type which maybe would help to enforce the cases. This class could also have methods to get minio ID and faban ID for the given test/experiment ID. 
In addition there are trial IDs: `<username>.<test name>.<test number>.<experiment number>.<trial number>`

**Comment by igormoreno** [comment]
Exactly. I think having abstractions (types) that captures that is essential. I've been looking into the code and I think there are hidden bugs associated with that. I don't think these things are consistently checked

**Comment by igormoreno** [comment]
Also as I said before

> you can create a BPMN model for an experiment using the test BPMN model but you never retrieve that experiment BPMN model because when you ask for it you actually get the test BPMN model retrieved using a testID generated from the experimentID.

We can sit down together and look into it if you want

**Comment by jeppe-style** [comment]
OK - I suggest then that these types go into a separate library since it is used by several services.

Regarding BPMN models - these are never changed between experiments. To not duplicate the files on minio they are accessed directly where they are stored for the test. However, the experiment manager (and other services) don't know about tests so therefore this conversion is there.

**Comment by VincenzoFerme** [comment]
For the types, I think the DSL is the candidate to handle all these mappings.

For the retrieval of test models for experiments, probably this has to be explained better in the code.
